src/vas.jl

```diff
@@ -10,36 +13,71 @@ is an array of values. Negative numbers mean the transition needs to
 take this many tokens from the state, meaning the state at those indices
 must be an integer at least that large. Positive numbers mean the
 transition places tokens into the state. Unlike chemical simulations,
-the rate doesn't depend on the number of combinations of species present.
+the rate need not depend on the number of combinations of species present.
 """
-struct VectorAdditionSystem
+struct VectorAdditionSystem{T <: Function}
```

Comment:
Good, good, catch.

Reply:
Full specialization for increased memory locality!